subgraphs/beefy-finance/src/mappings/vault.ts

```diff
@@ -0,0 +1,335 @@
+    vault.totalValueLockedUSD = inputToken.lastPriceUSD
+      .times(vault.inputTokenBalance.toBigDecimal())
+      .div(BIGINT_TEN.pow(inputToken.decimals as u8).toBigDecimal());
+  }
+
+  const outputSupply = vault.outputTokenSupply;
+  if (outputSupply && outputSupply != BIGINT_ZERO)
+    vault.outputTokenPriceUSD = vault.totalValueLockedUSD.div(
+      outputSupply.toBigDecimal()
+    );
+
+  const dailySnapshot = updateVaultDailySnapshot(block, vault);
+  if (
+    vault.dailySnapshots[vault.dailySnapshots.length - 1] != dailySnapshot.id
+  ) {
+    vault.dailySnapshots = vault.dailySnapshots.concat([dailySnapshot.id]);
+  }
+  const hourlySnapshot = updateVaultHourlySnapshot(block, vault);
+  if (
+    vault.hourlySnapshots[vault.hourlySnapshots.length - 1] != hourlySnapshot.id
+  ) {
+    vault.hourlySnapshots = vault.hourlySnapshots.concat([hourlySnapshot.id]);
+  }
+
+  vault.save();
+}
+
+export function getFees(
+  vaultId: string,
+  strategyContract: BeefyStrategy
+): string[] {
+  const strategistFee = new VaultFee("STRATEGIST_FEE-" + vaultId);
+  let call = strategyContract.try_STRATEGIST_FEE();
+  if (call.reverted) {
+    strategistFee.feePercentage = BIGDECIMAL_ZERO;
+  } else {
+    strategistFee.feePercentage = call.value.divDecimal(BIGDECIMAL_HUNDRED);
+  }
+  strategistFee.feeType = "STRATEGIST_FEE";
+  strategistFee.save();
+
+  const withdrawalFee = new VaultFee("WITHDRAWAL_FEE-" + vaultId);
+  call = strategyContract.try_withdrawalFee();
+  if (call.reverted) {
+    withdrawalFee.feePercentage = BIGDECIMAL_ZERO;
+  } else {
+    withdrawalFee.feePercentage = call.value.divDecimal(BIGDECIMAL_HUNDRED);
+  }
+  withdrawalFee.feeType = "WITHDRAWAL_FEE";
+  withdrawalFee.save();
+
+  const callFee = new VaultFee("MANAGEMENT_FEE-" + vaultId);
+  call = strategyContract.try_callFee();
+  if (call.reverted) {
+    callFee.feePercentage = BIGDECIMAL_ZERO;
+  } else {
+    callFee.feePercentage = call.value.divDecimal(BIGDECIMAL_HUNDRED);
+  }
+  callFee.feeType = "MANAGEMENT_FEE";
+  callFee.save();
+
+  const beefyFee = new VaultFee("PERFORMANCE_FEE-" + vaultId);
+  call = strategyContract.try_beefyFee();
+  if (call.reverted) {
+    beefyFee.feePercentage = BIGDECIMAL_ZERO;
+  } else {
+    beefyFee.feePercentage = call.value.divDecimal(BIGDECIMAL_HUNDRED);
+  }
+  beefyFee.feeType = "PERFORMANCE_FEE";
+  beefyFee.save();
+
+  return [strategistFee.id, withdrawalFee.id, callFee.id, beefyFee.id];
+}
+
+export function getVaultDailyRevenues(vault: Vault): BigDecimal[] {
+  let fee: VaultFee | null;
+  let lastSnapshot = VaultDailySnapshot.load(
+    vault.dailySnapshots[vault.dailySnapshots.length - 1]
+  );
+  if (lastSnapshot) {
+    let currentTotalRevenue = lastSnapshot.dailyTotalRevenueUSD;
+    let currentRevenueSupplySide = currentTotalRevenue;
+    let currentRevenueProtocolSide = BIGDECIMAL_ZERO;
+    for (let k = 0; k < vault.fees.length; k++) {
+      fee = VaultFee.load(vault.fees[k]);
```

Comment:
You are only updating the fees when the vault is created, this logic will fail if the fees are updated. You should either add handlers for this or do JSON RPC API calls whenever you need performance or withdrawal fees.